fusion/python/sample.py

```diff
@@ -0,0 +1,205 @@
+
+    Raises:
+        HTTPError: An error occurs from the HTTP request.
+    """
+    url = 'https://{0}{1}'.format(host, quote(path.encode('utf8')))
+    assert CLIENT_ID, "Please supply your client_id."
+    assert CLIENT_SECRET, "Please supply your client_secret."
+    data = urlencode({
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET,
+        'grant_type': GRANT_TYPE,
+    })
+    headers = {
+        'content-type': 'application/x-www-form-urlencoded',
+    }
+    response = requests.request('POST', url, data=data, headers=headers)
+    bearer_token = response.json()['access_token']
+    return bearer_token
+
+
+def request(host, path, bearer_token, url_params=None):
+    """Given a bearer token, send a GET request to the API.
+
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        bearer_token (str): OAuth bearer token, obtained using client_id and client_secret.
+        url_params (dict): An optional set of query parameters in the request.
+
+    Returns:
+        dict: The JSON response from the request.
+
+    Raises:
+        HTTPError: An error occurs from the HTTP request.
+    """
+    url_params = url_params or {}
+    url = 'https://{0}{1}'.format(host, quote(path.encode('utf8')))
+    headers = {
+        'Authorization': 'Bearer %s' % bearer_token,
+    }
+
+    print(u'Querying {0} ...'.format(url))
+
+    response = requests.request('GET', url, headers=headers, params=url_params)
+
+    return response.json()
+
+
+def search(bearer_token, term, location):
+    """Query the Search API by a search term and location.
+
+    Args:
+        term (str): The search term passed to the API.
+        location (str): The search location passed to the API.
+
+    Returns:
+        dict: The JSON response from the request.
+    """
+
+    url_params = {
+        'term': term.replace(' ', '+'),
+        'location': location.replace(' ', '+'),
+        'limit': SEARCH_LIMIT
+    }
+    return request(API_HOST, SEARCH_PATH, bearer_token, url_params=url_params)
+
+
+def get_business(bearer_token, business_id):
+    """Query the Business API by a business ID.
+
+    Args:
+        business_id (str): The ID of the business to query.
+
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, bearer_token)
+
+
+def query_api(term, location):
+    """Queries the API by the input values from the user.
+
+    Args:
+        term (str): The search term to query.
+        location (str): The location of the business to query.
+    """
+    bearer_token = obtain_bearer_token(API_HOST, TOKEN_PATH)
+
+    response = search(bearer_token, term, location)
+
+    businesses = response.get('businesses')
+
+    if not businesses:
+        print(u'No businesses for {0} in {1} found.'.format(term, location))
```

Comment:
Empty search does throw a 200, I thought this was hitting the business endpoint lookup (misread the code). Nevermind!